domain/schema/controller.go

```diff
@@ -357,6 +357,11 @@ CREATE TABLE model_metadata (
     model_type_id         INT,
     name                  TEXT NOT NULL,
     owner_uuid            TEXT NOT NULL,
+    -- target_agent_version describes the desired agent version that should be
+    -- being run in this model. It should not be considered "the" version that
+    -- is being run for every agent as each agent needs to upgrade to this
+    -- version.
+    target_agent_version  TEXT NOT NULL,
```

Comment:
This is a transient attribute in that once a model's actual version matches its target version, the attribute is not relevant till the next time an update is needed. My view is that I'd rather not have this populate the model_metadata table - such "workflow" related attributes should be in their own table. Doing model upgrades will have a bunch of related state that is used to control the upgrade workflow. In 3.x we even have a separate upgrade info doc to track it all
```
	Id               string         `bson:"_id"`
	PreviousVersion  version.Number `bson:"previousVersion"`
	TargetVersion    version.Number `bson:"targetVersion"`
	Status           UpgradeStatus  `bson:"status"`
	Started          time.Time      `bson:"started"`
	ControllersReady []string       `bson:"controllersReady"`
	ControllersDone  []string       `bson:"controllersDone"`
```
This even includes target version.
I would think we'd do something similar in dqlite.